FWCore/Framework/test/stubs/TestGlobalAnalyzers.cc

```diff
@@ -300,6 +301,86 @@ namespace edmtest {
       }
     };
 
+    class ProcessBlockIntAnalyzer : public edm::global::EDAnalyzer<edm::ProcessBlockCache<UnsafeCache>> {
+    public:
+      explicit ProcessBlockIntAnalyzer(edm::ParameterSet const& p) : trans_(p.getParameter<int>("transitions")) {
+        {
+          auto tag = p.getParameter<edm::InputTag>("consumesBeginProcessBlock");
+          if (not tag.label().empty()) {
```

Comment:
I copied this pattern from some other existing test module. My first thought is just to leave it as it is. This is repeated many dozens of places in tests in this PR. It would take a while to change and it is probably good enough. I never really liked complicated  conditional expressions much. But maybe I just need to get used to them. But your suggestion would work and I don't feel strongly either. I'll do it if you like. We could also write a new utility function, put it somewhere, and replace all these little blocks with one line function calls.
```
template <typename T>
void ifTagCallConsumes(ParameterSet&, EDConsumerBase*, const char* label, EDGetTokenT<T>&);
```

Reply:
Given that it is a copied pattern, let's keep it for now.